analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/ConditionEvaluatesToConstantBase.cs

```diff
@@ -62,13 +64,25 @@ public override void ExecutionCompleted()
 
         foreach (var operation in alwaysTrue)
         {
-            ReportIssue(Rule2589, operation, null);
+            ReportIssue(operation, true);
         }
         foreach (var operation in alwaysFalse)
         {
-            ReportIssue(Rule2589, operation, null);
+            ReportIssue(operation, false);
         }
 
         base.ExecutionCompleted();
     }
+
+    private void ReportIssue(IOperation operation, bool conditionEvaluation)
+    {
+        if (operation.Kind == OperationKindEx.IsNull)
+        {
+            ReportIssue(Rule2589, operation, null, MessageNullCoalescing);
+        }
+        else
+        {
+            ReportIssue(Rule2589, operation, null, string.Format(MessageBool, conditionEvaluation));
+        }
```

Comment:
```suggestion
        ReportIssue(Rule2589, operation, null, operation.Kind == OperationKindEx.IsNull ? MessageNullCoalescing : string.Format(MessageBool, conditionEvaluation));
```

Reply:
I find this a bit complicate to read. 
In the end I just extracted the coalescee outside of the call into a variable.